### ISSUE TITLE
[ENHANCEMENT] Allow to group mime types in `accept`

### ISSUE THREAD

**Issue by targos** [react-dropzone/react-dropzone#1189]
**Is your feature request related to a problem? Please describe.**

A side-effect of version 13.0.0 is that now each mime type defined in the `accept` option becomes a separate entry in the file picker.
`window.showOpenFilePicker` expects an array of entries where we can combine mime types and give a custom description, but AFAIK `react-dropzone` doesn't allow this and will always create an array entry for each mime type.

**Describe the solution you'd like**

One solution could be to let users pass an array of objects to `accept`, which would have the same shape as the `types` option of `window.showOpenFilePicker`:

```js
const dropzone = useDropzone({
  accept: [
    {
      description: 'All supported images',
      accept: {
        'image/jpeg': ['.jpg', '.jpeg'],
        'image/png': [],
      }
    },
    {
      description: 'JPEG images',
      accept: {
        'image/jpeg': ['.jpg', '.jpeg'],
      }
    },
    {
      description: 'PNG images',
      accept: {
        'image/png': [],
      }
    },
  ]
})
```

Result:
![image](https://user-images.githubusercontent.com/2352663/166674067-793bf1b7-30a7-4c80-882d-1cd98fed2f57.png)

**Describe alternatives you've considered**

Otherwise, a new option could be added for this, but it would be ambiguous if someone also passed `accept` at the same time.

**Comment by rolandjitsu** [comment]
This would be another breaking change I guess. But it's a valid request.

**Comment by thaiphan** [comment]
Yeah I prefer this if possible. In the meantime, I'm going to stick to v12.

**Comment by finnatsea** [comment]
This would our preference. Manually specifying is quite verbose.

**Comment by rolandjitsu** [comment]
@targos are you interested in making a PR with the enhancement?
